main/file.c

```diff
@@ -147,26 +167,65 @@ sub_status_t sub_init(sub_init_struct_t *init_struct, uint16_t transmit_periods[
     return SUB_OK;
 }
 
+static void set_transmit_state(bool state){
+
+    if(xSemaphoreTake(sub_struct.transmit_time_mutex, portMAX_DELAY) == pdTRUE){
+
+        sub_struct.transmit_state = state;
+
+        xSemaphoreGive(sub_struct.transmit_time_mutex);
+    }
+
+}
+
+static void transmit_timer_cb(TimerHandle_t xTimer){
+
+    //TODO: POZBYĆ SIĘ SEMAFORA Z CALLBACKA 
```

Comment:
Ogólnie podpinanie tych rtosowych timerów wewnątrz biblioteki nie jest najlepszym pomysłem, bo nie masz kontroli nad taskiem który go obsługuje oraz nad zadaniami które się na nim wykonują. Ja w lorze to zakodowałem dawno temu i nie miałem czasu tego zmienić. 

Ogólnie w związku z tym że masz tam swój task to bym po prostu przechowywał czas od ostatniego wysłania i sprawdzał czy już nie jest czas na nową paczkę.

Przy okazji wszystko związane z timer_state już nie będzie potrzebne